examples/minimal/packages/client-react-external-wallet/src/mud/customWalletClient.ts

```diff
@@ -0,0 +1,82 @@
+import type { WriteContractParameters, Chain, Account, WalletActions } from "viem";
+import { sendTransaction, writeContract } from "viem/actions";
+import { useAccount, useWalletClient, type UseWalletClientReturnType } from "wagmi";
+import pRetry from "p-retry";
+import { getNonceManager } from "@latticexyz/common";
+import { useMUDNetwork, type MUDNetwork } from "./NetworkContext";
+
+export const useMUD = () => {
+  const network = useMUDNetwork();
+  const { data: connectorWalletClient } = useWalletClient();
+  const { chainId } = useAccount();
+
+  let walletClient;
+  if (network.publicClient.chain.id === chainId && connectorWalletClient?.chain.id === chainId) {
+    // TODO: Should this be memoized?
+    // `walletClient = connectorWalletClient.extend(burnerActions);` is unnecessary for an external wallet
+    walletClient = connectorWalletClient.extend(setupObserverActions(network.onWrite));
+  }
+
+  return { network, walletClient };
+};
+
+export type WalletClient = NonNullable<UseWalletClientReturnType["data"]>;
+
+// See @latticexyz/common/src/sendTransaction.ts
+const burnerActions = (client: WalletClient): Pick<WalletActions<Chain, Account>, "sendTransaction"> => {
+  // TODO: Use the `debug` library once this function has been moved to the `common` library.
+  const debug = console.log;
+
+  return {
+    sendTransaction: async (args) => {
```

Comment:
This functions similarly to `mud_writeContract` and `mud_sendTransaction` through the viem client interactions, with the exception of transaction simulation. This approach is consistent with the standard viem action behavior, which requires users to initiate the simulation separately when needed.